src/components/NewVault.tsx

```diff
@@ -0,0 +1,34 @@
+import { useVaultStore } from 'store/vaults';
+
+const NewVault = ({ id }: { id: string }) => {
+  const {
+    vaultGovernedParams,
+    vaultLoadingErrors,
+    vaultManagers,
+    vaultMetrics,
+    prices,
+    priceErrors,
+  } = useVaultStore();
+
+  const manager = vaultManagers.get(id);
+  const metrics = vaultMetrics.get(id);
+  const params = vaultGovernedParams.get(id);
+  const brand = metrics?.retainedCollateral?.brand;
+  const price = brand && prices.get(brand);
+
+  const error = vaultLoadingErrors.get(id) || (brand && priceErrors.get(brand));
+
+  const isLoading =
+    !(manager && metrics && params && price) || error != undefined;
```

Comment:
consider simplifying the logical expression:
```suggestion
  const isReady =
    manager && metrics && params && price && error === undefined;
```

Reply:
Done